development.md

```diff
@@ -0,0 +1,79 @@
+
+The contributions guideline provides details on copyright and process.  This document focuses on technical toolchain and local development.
+
+## Toolchain
+
+The follow are a list of tools needed in order to build, run, test and debug locally.
+
+* make - used to build and test locally
+* go - KUDO is written in Go, we use go 1.13+ style of development
+* [goreleaser](https://goreleaser.com/) - is used for the release process
+* docker
+* [kind](https://kind.sigs.k8s.io/) and/or [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/)
+* [jq](https://stedolan.github.io/jq/)
+* [yq](https://github.com/mikefarah/yq)
+* Unix commands:  awk, sed, cut, tar, sleep, curl, tee
+* git
+* kubectl
+* [ngrok](https://ngrok.com/) - This is currently needed for full debugging. We are looking at alternatives. It does require signing up for this service.
+
+## Running Locally
+
+These details will show using kind as a local kubernetes cluster.  These instructions should work with minikube as well.
+You have 2 options when running locally.
+1. you can run the kudo controller and webhooks in the cluster or
+2. you can run the kudo controller and webhooks outside the cluster for debugging
+
+When running locally we assume that you will not be using a CertManager thus initialization is with `--unsafe-self-signed-webhook-ca`.
+
+### Running KUDO locally in Cluster
+
+When running locally in the cluster, we use the kudo CLI code to deploy the manager. You will either need to launch a released kudo manager which means you have to pick a version.  Example:
+`go run cmd/kubectl-kudo/main.go init --unsafe-self-signed-webhook-ca --version 0.15.0 -w`
+
+In this example, we are choosing a released version of kudo which is compatiable with the current cli version.  The `-w` is to wait until this is running.
+This command will initialize the cluster with all the prerequisite configurations for serviceaccounts, namespaces, etc.
+
+If you want to run a non-released version of kudo, then build the latest code with docker.  `make docker-build`.  This process will end with something like:
+```
+Successfully built 2eef6479680e
+Successfully tagged kudobuilder/controller:cbe6e68c270221a69b7e1f48f8f22fc39ade5c47
+```
+Retag and push this image:
+```
+ docker tag  kudobuilder/controller:cbe6e68c270221a69b7e1f48f8f22fc39ade5c47 kensipe/controller:latest
+ docker push kensipe/controller:latest
+```
+
+Now run locally with `go run cmd/kubectl-kudo/main.go init --unsafe-self-signed-webhook-ca --kudo-image kensipe/controller:latest -w`
+
+### Debugging KUDO Running locally outside the Cluster
+
+The most significant challenge for running kudo locally outside the cluster is kudo v0.14.0+ requires a webhook.  This webhook configuration requires kube to make a callback to the webhook.  This callback is accomplished via an https call requiring valid certificates to work.  Our current solution to this is to use ngrok which was created for this purpose.
```

Comment:
kudo -> KUDO

Reply:
kube -> kubernetes

Reply:
link to the official docs somewhere https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/